File: VB/CustomRecurrenceFormWinFormSample/Form1.vb
Imports System
Imports System.Windows.Forms

Namespace CustomRecurrenceFormWinFormSample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            ' Handle the EditAppointmentFormShowing event to invoke a custom form instead of the default appointment editing form.
            AddHandler Me.schedulerControl1.EditAppointmentFormShowing, AddressOf schedulerControl1_EditAppointmentFormShowing
            AddHandler Me.Shown, AddressOf Form1_Shown
        End Sub

        Private Sub schedulerControl1_EditAppointmentFormShowing(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.AppointmentFormEventArgs)
            Dim myForm As New MyAppointmentRecurrenceForm(schedulerControl1, e.Appointment)
            myForm.ShowDialog()
            e.Handled = True
        End Sub

        Private Sub Form1_Shown(ByVal sender As Object, ByVal e As EventArgs)
            Me.schedulerControl1.CreateAppointment(False, True)
        End Sub
    End Class
End Namespace

File: CS/CustomRecurrenceFormWinFormSample/Form1.cs
using System;
using System.Windows.Forms;

namespace CustomRecurrenceFormWinFormSample
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            // Handle the EditAppointmentFormShowing event to invoke a custom form instead of the default appointment editing form.
            this.schedulerControl1.EditAppointmentFormShowing += schedulerControl1_EditAppointmentFormShowing;
            this.Shown += Form1_Shown;
        }
        
        void schedulerControl1_EditAppointmentFormShowing(object sender, DevExpress.XtraScheduler.AppointmentFormEventArgs e)
        {
            MyAppointmentRecurrenceForm myForm = new MyAppointmentRecurrenceForm(schedulerControl1, e.Appointment);
            myForm.ShowDialog();
            e.Handled = true;
        }

        void Form1_Shown(object sender, EventArgs e)
        {
            this.schedulerControl1.CreateAppointment(false, true);
        }
    }
}
